Metalama.Framework.Engine/Validation/ValidationResult.cs

```diff
@@ -5,4 +5,4 @@
 
 namespace Metalama.Framework.Engine.Validation;
 
-internal sealed record ValidationResult( ImmutableArray<ReferenceValidatorInstance> ExternallyVisibleValidations, ImmutableUserDiagnosticList Diagnostics );
\ No newline at end of file
+internal sealed record ValidationResult( ImmutableArray<ValidatorInstance> Validations, ImmutableArray<ReferenceValidatorInstance> ExternallyVisibleValidations, ImmutableUserDiagnosticList Diagnostics );
```

Comment:
Collecting all declaration validators is useless. A `bool` property `HasDeclarationValidator` is enough.